addons/product/models/product_pricelist.py

```diff
@@ -168,7 +175,7 @@ def _compute_price_rule(self, products, qty, uom=None, date=False):
             date = fields.Datetime.now()
 
         # Fetch all rules potentially matching specified products/templates/categories and date
-        rules = self._get_applicable_rules(products, date)
+        rules = self._get_applicable_rules(products, date) if self else None
```

Comment:
consistent type of variable content please :pray:
`type(rules) = random(None, recordset)`

(It might be cleaner to make `_get_applicable_rules` return an empty recordset when self is empty, but not 100% sure)